core/system/src/pool_queue.c

```diff
@@ -419,3 +419,160 @@ uvisor_pool_slot_t uvisor_pool_queue_find_first(uvisor_pool_queue_t * pool_queue
 
     return slot;
 }
+
+bool uvisor_pool_is_sane(uvisor_pool_t * pool, void * array,
+                         size_t stride, size_t num)
+{
+    /* Invalid num can never be sane */
+    if (num == 0 || num >= UVISOR_POOL_MAX_VALID) {
+        return false;
+    }
+
+    /* Check magic value */
+    if (pool->magic != UVISOR_POOL_MAGIC) {
+        return false;
+    }
+
+    /* Check pool size */
+    if (pool->num != num) {
+        return false;
+    }
+
+    /* Check array pointer */
+    if (pool->array != array) {
+        return false;
+    }
+
+    /* Check stride */
+    if (pool->stride != stride) {
+        return false;
+    }
+
+    /* Cannot check sanity if lock is held by others */
+    if (pool->spinlock.acquired) {
```

Comment:
Should we use try_lock instead to avoid race conditions in reading the spinlock's acquired state?

Reply:
Shall we lock inside or outside?

Reply:
Outside seems best